File: my_project_typescript/medium_problems_typescript/from51to100/pathSum.ts

 class TreeNode {
     val: number
      left: TreeNode | null
     right: TreeNode | null
      constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
          this.val = (val===undefined ? 0 : val)
          this.left = (left===undefined ? null : left)
          this.right = (right===undefined ? null : right)
      }
  }
 

 function pathSum(root: TreeNode | null, targetSum: number): number[][] {

    if (!root){
        return []
    }
    if (root.left === null && root.right === null){
        if (targetSum === root.val){
            return [[root.val]]
        }
        else {
            return []
        }
    }

    let a: Array<Array<number>> = pathSum(root.left, targetSum - root.val).concat(pathSum(root.right, targetSum-root.val))

    let temp: Array<Array<number>> = []

    for (let i of a){
        temp.push([root.val].concat(i))
    }
    return temp
};

File: src/my_project/interviews/top_150_questions_round_17/path_sum.py
from typing import List, Union, Collection, Mapping, Optional
from abc import ABC, abstractmethod

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:

    def hasPathSum(self, root: TreeNode, targetSum: int) -> bool:

        if not root:
            return False 
        if not root.left and not root.right and root.val == targetSum:
            return True 
        else:
            temp_target = targetSum - root.val 
            return self.hasPathSum(root.left, temp_target) \
                   or self.hasPathSum(root.right, temp_target)